### ISSUE TITLE
Missing include in types.h

### ISSUE THREAD

**Issue by yo35** [Geode-solutions/OpenGeode#843]
Hi,

An error is raised when `geode/basic/types.h` is included in a .cpp file:

```
In file included from /home/yoann/DevPro/Total_Flowers/REMMapping/include/remmapping/surface_tools.h:3,
                 from /home/yoann/DevPro/Total_Flowers/REMMapping/src/remmapping/surface_tools.cpp:1:
/opt/yoann/DevPro/Total_Flowers/install-dirs/geode/include/geode/basic/types.h:57:20: error: ‘array’ in namespace ‘std’ does not name a template type
   57 |     constexpr std::array< T, sizeof...( Args ) > to_array( Args&&... args )
      |                    ^~~~~
/opt/yoann/DevPro/Total_Flowers/install-dirs/geode/include/geode/basic/types.h:31:1: note: ‘std::array’ is defined in header ‘<array>’; did you forget to ‘#include <array>’?
   30 | #include <absl/strings/string_view.h>
  +++ |+#include <array>
   31 | 
```

The compiler is GCC 11.4, the OpenGeode version is 14.8.0.

**Comment by BotellaA** [comment]
Do not hesitate to contribute with PR.This will be fixed by #844

**Comment by yo35** [comment]
Thanks @BotellaA
